### ISSUE TITLE
App directory server component no working when not wrapped 

### ISSUE THREAD

**Issue by titouv** [QuiiBz/next-international#60]
Test with the new version in app directory in server component

I've encountered issues when a server component is not wrap using another function in the file. It works when its wrap by either a async or non-async component.
```ts
import { getI18n} from '../../locales/server';

async function Content() {
  const t = await getI18n();
  return (
    <></>
  );
}

export default function Home() {
  return <Content />;
}
```

Is this wrapping condition to work a bug or an expected nextjs app directory behavior (if yes why could be interesting)

**Comment by QuiiBz** [comment]
Hey, what's the error you're getting?

**Comment by titouv** [comment]
Sorry, i get the following `Error: No locale set, make sure to use the I18nProviderServer component`

https://github.com/titouv/next-international-repro-bug

Here is the link to the repro, the root path is failing with the previous errors but the root with async wrapper and the root with the non async wrapper works perfectly. Maybe there is something I don't understand about NextJS behavior

**Comment by titouv** [comment]
With further testing it appears the error also appears when navigating to a page, a simple reload removes the issue but the navigation make this error happen even though its wrapped

**Comment by titouv** [comment]
Seems in the repro and in the example from the repo (at least on my side), the getLocale function is run before the setLocale, which is thus problematic. If this can give an hint

**Comment by QuiiBz** [comment]
Looks like `cache` isn't the ideal solution to save the current locale for a given session. I'll try to use server contexts instead.

**Comment by QuiiBz** [comment]
I've published [0.5.2](https://github.com/QuiiBz/next-international/releases/tag/0.5.2) which resolves the issue on my side. Could you try again and open another issue if needed? Thanks.

**Comment by titouv** [comment]
Thanks for resolving, the issue. I experimented with server context on my side also, but setting headers is a better solutions. Thanks 👍
